src/HyperFormula.ts

```diff
@@ -4104,6 +4104,42 @@ export class HyperFormula implements TypedEmitter {
     return this._exporter.exportScalarOrRange(internalCellValue)
   }
 
+  /**
+   * Return a list of named expressions used by a formula.
+   *
+   * @param {string} formulaString - A formula in a proper format, starting with `=`.
+   *
+   * @throws [[ExpectedValueOfTypeError]] if any of its basic type arguments is of wrong type.
+   * @throws [[NotAFormulaError]] when the provided string is not a valid formula (i.e., doesn't start with `=`).
+   *
+   * @example
+   * ```js
+   * const hfInstance = HyperFormula.buildEmpty();
+   *
+   * // returns a list of named expressions used by a formula
+   * // for this example, returns ['foo', 'bar']
+   * const namedExpressions = hfInstance.getNamedExpressionsFromFormula('=foo+bar*2');
+   * ```
+   *
+   * @category Helpers
+   */
+  public getNamedExpressionsFromFormula(formulaString: string): string[] {
+    validateArgToType(formulaString, 'string', 'formulaString')
+    const { ast, dependencies } = this.extractTemporaryFormula(formulaString)
+
+    if (ast === undefined) {
+      throw new NotAFormulaError()
+    }
+
+    const namedExpressionDependencies = dependencies
+      .filter(dep => dep instanceof NamedExpressionDependency)
+      .map(namedExpr => (namedExpr as NamedExpressionDependency).name)
```

Comment:
is it possible to avoid doing a `as NamedExpressionDependency` here? perhaps we can do some narrowing etc

Reply:
Good point. I used type narrowing in the filter method